source/ShipInfoPanel.cpp

```diff
@@ -463,32 +473,93 @@ void ShipInfoPanel::DrawWeapons(const Rectangle &bounds)
 	Color topColor;
 	bool hasTop = false;
 	auto layout = Layout(static_cast<int>(LABEL_WIDTH), Truncate::BACK);
+
+	// First Calculate how many pages are needed. One is always there.
+	pages = 1;
+	int gunIndex = 0;
+	int turretIndex = 0;
+	int rowsPerPage = static_cast<int>((sprite->Height() * scale) / 10.) - 1;
+	int offset = 0;
+	while(pageNeeded)
+	{
+		if(gunIndex < gunRows)
+		{
+			if((gunRows - gunIndex) * 20. >= height)
+				gunIndex += (rowsPerPage / 2);
+			else
+			{
+				offset = gunRows - gunIndex;
+				gunIndex = gunRows;
+			}
+		}
+		if(gunIndex >= gunRows && turretIndex < turretRows)
+		{
+			if(offset)
+				if((turretRows - turretIndex + offset) * 20. + 10. >= height)
+					turretIndex += (rowsPerPage / 2) - (offset + 1);
+				else
+					turretIndex = turretRows;
+			else
+			{
+				if((turretRows - turretIndex) * 20. >= height)
+					turretIndex += (rowsPerPage / 2);
+				else
+					pageNeeded = false;
+			}
+		}
+		if(turretIndex >= turretRows && gunIndex >= gunRows)
+			pageNeeded = false;
+		pages++;
+		offset = 0;
+	}
+	weaponsRight.clear();
+	weaponsLeft.clear();
+	indecesRight.clear();
+	indecesLeft.clear();
+	int index = 0;
 	for(const Hardpoint &hardpoint : ship.Weapons())
 	{
+		if(hardpoint.GetPoint().X() >= 0.)
+		{
+			indecesRight.emplace_back(index);
+			weaponsRight.emplace_back(&hardpoint);
+		}
+		else
+		{
+			indecesLeft.emplace_back(index);
+			weaponsLeft.emplace_back(&hardpoint);
+		}
+		index++;
+	}
+
+	auto drawElements = [&] (std::vector<const Hardpoint*> &weaponList, int weaponIndex, bool right) {
```

Comment:
Lambdas are uppercased, like functions: `DrawElements`. Also a space between `Hardpoint` and `*`. And the { on a new line.

Reply:
okidoki